src/ORM/DataList.php

```diff
@@ -1128,41 +1136,59 @@ private function fetchEagerLoadHasOne(
             if (is_array($parentData)) {
                 // $parentData represents a record in this DataList
                 $hasOneID = $parentData[$hasOneIDField];
-                $fetchedIDs[] = $hasOneID;
-                $addTo[$hasOneID][] = $parentData['ID'];
+
+                if ($hasOneID > 0) {
+                    $hasOneClass = $schema->baseDataClass($parentData[$hasOneClassField] ?? $relationDataClass);
```

Comment:
Why are we getting the base data class specifically?
And why do we fall back on `$relationDataClass`?

Reply:
Because the "HasOneClass" field could contain any sub class, with this will only do one fetch.

The fallback is for normal has_one relations.

Reply:
I removed the baseClass check.

The idea was to remove multiple Queries for the same base DataObject, but as it would trigger lazyLoad queries afterwards this was a bad idea.